### PR TITLE
Add retry functionality

### PR DESCRIPTION
**Pull request by alexmuller**
The hellomama-registration app retries HTTP/HTTPS requests - this should be the responsibility of this project.

I've created a base API class as I think there's likely to be a lot of duplication between the different clients.

Not sure whether this is too complicated or not, feedback welcome.

### CODE REVIEW

File: seed_services_client/identity_store.py [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
 
-    def __init__(self, auth_token, api_url, session=None):
-        headers = {'Authorization': 'Token ' + auth_token}
-        if session is None:
-            session = JSONServiceClient(url=api_url,
-                                        headers=headers)
-        self.session = session
+    def __init__(self, auth_token, api_url, session=None, **kwargs):
+        super(IdentityStoreApiClient, self).__init__(
+            auth_token, api_url, session=None, **kwargs)
```
Comment (KaitCrawford) [inline seed_services_client/identity_store.py]: Wouldn't this set `session` to `None` regardless of what this was called with?

File: seed_services_client/seed_services.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    def __init__(self, auth_token, api_url, session=None, retry=False):
+        headers = {'Authorization': 'Token ' + auth_token}
+        if session is None:
+            session = JSONServiceClient(url=api_url,
+                                        headers=headers)
+        self.session = session
+
+        if retry:
+            http = HTTPAdapter(max_retries=5)
+            https = HTTPAdapter(max_retries=5)
```
Comment (KaitCrawford) [inline seed_services_client/seed_services.py]: Is there some way we could have `max_retries` be configurable? like as one of the arguments or something?

**Review by KaitCrawford — COMMENTED** [review]
:cake: Thanks for doing this @alexmuller !! It's looking great!
I've left two small queries for you

File: seed_services_client/identity_store.py (hunk 1 above)
Comment (alexmuller) [reply seed_services_client/identity_store.py]: Ah good catch, thanks! Fixed in 652303363977a2af933ae127a1b3b9f5697f8d08.

File: seed_services_client/seed_services.py (hunk 2 above)
Comment (alexmuller) [reply seed_services_client/seed_services.py]: 👍 Added in 7d0b235af35aac10c3e661f7c110aeae5c17e314.

File: seed_services_client/identity_store.py (hunk 1 above)
Comment (alexmuller) [reply seed_services_client/identity_store.py]: Also updated the first identity store test to make sure it doesn't happen again.

**Review by rudigiesler — APPROVED** [review]
:+1: . I'm assuming that we're also going to change over the clients for the other services to also use this new base class?

**Comment by alexmuller** [comment]
I think it makes sense to do as they're worked on but I didn't want to break them all in one go :)

**Pull request merged**